Analytics/AnalyticsTests/PluginTests/LifecycleTrackingPluginTests.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  LifecycleTrackingPluginTests.swift
+//  AnalyticsTests
+//
+//  Created by Satheesh Kannan on 11/03/25.
+//
+
+import UIKit
+import XCTest
+@testable import Analytics
+
+final class LifecycleTrackingPluginTests: XCTestCase {
+    var analyticsMock: AnalyticsClient?
+    var plugin: LifecycleTrackingPlugin!
+    
+    override func setUp() {
+        super.setUp()
+        analyticsMock = MockProvider.clientWithMemoryStorage
+        plugin = LifecycleTrackingPlugin()
+    }
+    
+    override func tearDown() {
+        analyticsMock = nil
+        plugin = nil
+        super.tearDown()
+    }
+    
+    func test_application_installed_event() async {
+        print("Given an analytics configuration allows lifecycle tracking")
+        analyticsMock?.configuration.trackApplicationLifecycleEvents = true
+        
+        print("When the plugin setup with the given analytics configuration")
+        guard let analyticsMock else { XCTFail("No disk client"); return }
+        plugin.setup(analytics: analyticsMock)
+        
+        print("Then the app version should be initialized")
+        XCTAssertNotNil(plugin.appVersion)
+        
+        print("Then the plugin should track installation event")
+        guard let eventName = await readCurrentEventName() else { XCTFail("No event recorded"); return }
+        XCTAssert(eventName == LifecycleEvent.applicationInstalled.rawValue)
+    }
+    
+    func test_application_opened_event() async {
+        print("Given an analytics configuration allows lifecycle tracking")
+        analyticsMock?.configuration.trackApplicationLifecycleEvents = true
+        guard let analyticsMock else { XCTFail("No disk client"); return }
+        plugin.setup(analytics: analyticsMock)
+        
+        print("When the app become active")
+        plugin.onBecomeActive()
+        
+        print("Then the plugin should track application opened event")
+        guard let eventName = await readCurrentEventName() else { XCTFail("No event recorded"); return }
+        XCTAssert(eventName == LifecycleEvent.applicationOpened.rawValue)
+    }
+    
+    func test_application_backgrounded_event() async {
+        print("Given an analytics configuration allows lifecycle tracking")
+        analyticsMock?.configuration.trackApplicationLifecycleEvents = true
+        guard let analyticsMock else { XCTFail("No disk client"); return }
+        plugin.setup(analytics: analyticsMock)
+        
+        print("When the app become active")
+        plugin.onBackground()
```

Comment:
I think it should be like this:
```suggestion
        print("When the app moves to background")
        plugin.onBackground()
```